plugins/kubernetes-backend/src/cluster-locator/index.ts

```diff
@@ -38,13 +41,17 @@ class CombinedClustersSupplier implements KubernetesClustersSupplier {
 
 export const getCombinedClusterSupplier = (
   rootConfig: Config,
+  discovery: PluginEndpointDiscovery,
   refreshInterval: Duration | undefined = undefined,
 ): KubernetesClustersSupplier => {
+  const catalogClient = new CatalogClient({ discoveryApi: discovery });
```

Comment:
If we just use discovery to create a CatalogClient, why don't we just pass the catalog client we already create from discovery [here]([standaloneApplication.ts](https://github.com/backstage/backstage/pull/11789/files#diff-d343f848736906b291d4a16a7bbc66b327fccd20b46bbffaa558e404c670f09e))

Reply:
👍 and we also already pass a client into the original builder, so should be available in all cases now. Maybe the added discovery parameter can be removed everywhere then